src/cli/cli.ts

```diff
@@ -0,0 +1,569 @@
+  name: 'list-arns-auctions',
+  description: 'Get all ArNS auctions',
+  options: paginationOptions,
+  action: (o) =>
+    readIOFromOptions(o)
+      .getArNSAuctions(paginationParamsFromOptions(o))
+      .then((result) =>
+        result.items.length ? result : { message: 'No auctions found' },
+      ),
+});
+
+makeCommand<AuctionPricesOptions>({
+  name: 'get-arns-auction-prices',
+  description: 'Get ArNS auction prices',
+  options: arNSAuctionPricesOptions,
+  action: async (options) => {
+    options.type ??= 'lease';
+    if (options.type !== 'lease' && options.type !== 'permabuy') {
+      throw new Error(`Invalid type. Valid types are: lease, permabuy`);
+    }
+
+    const result = await readIOFromOptions(options).getArNSAuctionPrices({
+      name: requiredNameFromOptions(options),
+      type: options.type,
+      intervalMs:
+        options.intervalMs !== undefined ? +options.intervalMs : undefined,
+      timestamp:
+        options.timestamp !== undefined ? +options.timestamp : undefined,
+      years: options.years !== undefined ? +options.years : undefined,
+    });
+    return result ?? { message: `No auction prices found` };
+  },
+});
+
+makeCommand<EpochOptions>({
+  name: 'get-epoch',
+  description: 'Get epoch data',
+  options: epochOptions,
+  action: (o) =>
+    readIOFromOptions(o)
+      .getEpoch(epochInputFromOptions(o))
+      .then(
+        (result) => result ?? { message: `No epoch found for provided input` },
+      ),
+});
+
+makeCommand({
+  name: 'get-current-epoch',
+  description: 'Get current epoch data',
+  action: (options) => readIOFromOptions(options).getCurrentEpoch(),
+});
+
+makeCommand({
+  name: 'get-prescribed-observers',
+  description: 'Get prescribed observers for an epoch',
+  options: epochOptions,
+  action: (o) =>
+    readIOFromOptions(o)
+      .getPrescribedObservers(epochInputFromOptions(o))
+      .then((result) =>
+        result.length ? result : { message: `No observers found for epoch` },
+      ),
+});
+
+makeCommand({
+  name: 'get-prescribed-names',
+  description: 'Get prescribed names for an epoch',
+  options: epochOptions,
+  action: (o) =>
+    readIOFromOptions(o)
+      .getPrescribedNames(epochInputFromOptions(o))
+      .then((result) =>
+        result.length ? result : { message: `No names found for epoch` },
+      ),
+});
+
+makeCommand({
+  name: 'get-observations',
+  description: 'Get observations for an epoch',
+  options: epochOptions,
+  action: (o) => readIOFromOptions(o).getObservations(epochInputFromOptions(o)),
+});
+
+makeCommand({
+  name: 'get-distributions',
+  description: 'Get distributions for an epoch',
+  options: epochOptions,
+  action: (o) =>
+    readIOFromOptions(o).getDistributions(epochInputFromOptions(o)),
+});
+
+makeCommand({
+  name: 'get-token-cost',
+  description: 'Get token cost',
+  options: tokenCostOptions,
+}).action(async (_, command) => {
+  await runCommand<GetTokenCostOptions>(command, async (options) => {
```

Comment:
nit: could use its own file. other medium sized read functions could share a `readCommands.ts`. commands that have actions that are 3 lines or less can stay here